esp-hal/src/aes/mod.rs

```diff
@@ -374,12 +439,16 @@ pub mod dma {
         C: ChannelTypes,
         C::P: AesPeripheral,
     {
+        /// Writes the encryption key to the AES hardware, checking that its
+        /// length matches expected constraints.
         pub fn write_key(&mut self, key: &[u8]) {
```

Comment:
Shouldn't this also take a `Key` instead of a slice?

Reply:
Done for both requests, thank you!